src/shipment/controller.js

```diff
@@ -155,6 +155,93 @@ async function getAll(_req, res) {
   );
 }
 
+async function getPending(_req, res) {
+  ExpressHandler(
+    async () => {
+      const shipments = await Shipment.find({
+        $or: [
+          {
+            // shipment.deliveryMethod === "CARRIER" &&
+            // (!shipment.deliverySpeed || !shipment.trackingNumber || shipment.cost == null | undefined)
+            $and: [
+              {
+                deliveryMethod: "CARRIER",
+              },
+              {
+                $or: [
+                  {
+                    $or: [
+                      { deliverySpeed: { $exists: false } },
+                      { deliverySpeed: { $eq: undefined } },
+                      { deliverySpeed: { $eq: null } },
+                    ],
+                  },
+                  {
+                    $or: [
+                      { trackingNumber: { $exists: false } },
+                      { trackingNumber: { $eq: undefined } },
+                      { trackingNumber: { $eq: null } },
+                    ],
+                  },
+                  {
+                    $or: [
+                      { cost: { $exists: false } },
+                      { cost: { $eq: undefined } },
+                      { cost: { $eq: null } },
+                    ],
+                  },
+                ],
+              },
+            ],
+          },
+          {
+            // If( PICKUP or DROPOFF ) &  customerHandoffName not set
+            $and: [
+              {
+                $or: [
+                  {
+                    deliveryMethod: "PICKUP",
+                  },
+                  {
+                    deliveryMethod: "DROPOFF",
+                  },
+                ],
+              },
+              {
+                $or: [
+                  {
+                    customerHandoffName: { $eq: undefined },
```

Comment:
I think this is redundant. Null works for a value that doesn't exist or is just not set yet. At least in my testing it did.

Reply:
What @jbolles94 said is correct. MongoDB casts `undefined` to `null`.

Ref: https://mongoplayground.net/p/m2yXwDPVldi

Reply:
Sounds good!